ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -284,8 +285,11 @@ private void registerQuery(
       }
     }
     allLiveQueries.add(query);
+    // For the CSU metrics we need to have initialized first. It seems like initialize could throw errors though
+    // so we probably want to notify the other listeners first
```

Comment:
if you rebase on top of https://github.com/confluentinc/ksql/pull/7627 (once its merged) you wont need this